doc/builders/images/binarycache.section.md

```diff
@@ -0,0 +1,32 @@
+# pkgs.mkBinaryCache {#sec-pkgs-binary-cache}
+
+`pkgs.mkBinaryCache` is a function for creating Nix flat-file binary caches. Such a cache exists as a directory on disk, and can be used as a Nix substituter by passing `--substituter file:///path/to/cache` to Nix commands.
+
+Nix packages are most commonly shared between machines using [HTTP, SSH, or S3](https://nixos.org/manual/nix/stable/package-management/sharing-packages.html), but a flat-file binary cache can still be useful in some situations. For example, you can copy it directly to another machine, or make it available on a network file system. It can also be a convenient way to make some Nix packages available inside a container via bind-mounting, when you want to efficiently run Nix builds inside a container without doing a writable mount of the host's `/nix` directory.
+
+## Example
+
+The following derivation will construct a flat-file binary cache containing the closure of `hello`.
+
+```nix
+mkBinaryCache {
+  rootPaths = [hello];
+}
+```
+
+- `rootPaths` specifies a list of root derivations. The transitive closure of these derivations will be copied into the cache.
```

Comment:
````suggestion
## Arguments

- `rootPaths` specifies a list of root derivations. The transitive closure of these derivations will be copied into the cache.

## Examples

The following derivation will construct a flat-file binary cache containing the closure of `hello`.

```nix
mkBinaryCache {
  rootPaths = [hello];
}
```
````

Let's keep the examples together, I guess?

Reply:
Putting the arguments in their own section above is different from how all the other documentation does it; please let's keep it as-is and maintain consistency.

Reply:
I agree in principle with consistency, but now you interleave examples and arguments. Without the sub-heading its not clear that there is an argument listing.

Reply:
I totally feel you, the PR cycle can really be exhausting.

The suggestions are just suggestions, and we're trying to find the best solution. There is no authoritative stance on the topic, this is why there is so much case-by-case bickering.

My basis of argument is probably weak from your POV: just because it's done in a certain way it doesn't mean it's good. I'd rather start improving right now and do it in a principled manner on fresh material, define a rule while at it, and point people to it whenever they touch old material.

You don't have to agree, you own this code after all. Your approach is not worse in terms of results. The reason I don't favor it is that Big Bang changes are tedious to execute for a few reasons and a focused effort in that direction will never be prioritised because the overall payoff is small. That's the only reason I prefer temporary (even longer-term) inconsistency due to ongoing improvements over almost guaranteed permanent non-improvement.

Reply:
> There is no authoritative stance on the topic

Is there work being done on guidelines?

Reply:
Well, putting aside the Big Bang debate on which you already know my position -- in this particular case I don't even agree that your alternative is better. I think it's more verbose, and doesn't flow as well since usually you want to know the details on a function argument after seeing how it's used.

I'd totally support an authoritative style guide though. Anyway, I couldn't quite tell from your message -- is that an insist or not insist? :)

Reply:
> The suggestions are just suggestions

No insisting to be found here. :)

Reply:
Okay then, with apologies I'm going to leave it as-is and resolve this comment.

I might have said this before, but if you'd accept some advice -- I really don't think this is an appropriate or effective way to go about obtaining changes to the docs. If there's going to be a change to how the docs are written, then that deserves a full and public discussion process. At a minimum, the people who wrote the docs the way they currently are deserve a chance to weigh in. If I were a docs contributor and found that someone had forced through a convention change via drive-by on someone else's PR, without review, then I might be upset. Especially if it wasn't universally obvious that the change is an improvement.

On the positive side, you can be the one to drive the changes you want to see! The first obvious step would be to write the authoritative style guide and get it discussed and approved. And if that goes well, the next step would be to write incremental PRs to transform the docs. Maybe add a link to the style guide to the GitHub PR message, or make a git hook that fires when people touch the docs, or a even come up with a linter. That's the sort of process I'm accustomed to seeing on big open-source projects.

As always I appreciate the review and wish you luck on your documentation-improving missions.

Reply:
Thanks for the feedback, I really appreciate it, and agree with everything you wrote in that regard. I do work on the guidelines, and I'm trying to take every stake holder on board with that. It's a constant struggle to strike the right balance between incremental progress and large-scale changes.